### DISCUSSION TITLE
Importing an ADIF file does not use the MY_GRIDSQUARE value but uses my profile grid square

### DISCUSSION THREAD

**Discussion by AD7Djoe** [magicbug/Cloudlog#1228] [General]
I imported an ADIF file that I had a bunch of contacts I made from the gridsquare KO09qs.

However, when I view the contacts it shows my usual station gridsquare KP20dl.

When importing an ADIF file, if the MY_GRIDSQUARE value is specifically declared, this value should be used for the contact.  This is what LOTW uses.

Repro Steps:

1.  My station grid square in Cloudlog is KP20dl for OG7D.
2. I import to my OG7D the following (note my gridsquare for the contact was KO09qs, which is correct for the island I was on:
<CALL:6>PE1CMO<QSO_DATE:8>20211012<TIME_ON:6>141908<BAND:4>13cm<MODE:3>SSB<SAT_MODE:3>SSB<SAT_NAME:6>QO-100<PROP_MODE:3>SAT<OPERATOR:4>OG7D<STATION_CALLSIGN:4>OG7D<MY_GRIDSQUARE:6>KO09qs<COMMENT:36>PE1CMO Tue, 12 Oct 2021 14:19:08 GMT<eor>
3.  Navigate to Clouglog -> Logbook -> PE1CMO -> click PE1CMO
4. In the QSO Data dialog, click Station Location.
5. Look at Station Gridsquare

Results:

Station Gridsquare  KP20DL

Expected:

Station Gridsquare KO09qs.

**Comment by AndreasK79** [comment]
I agree. I just checked the code, and at first it did look ok. It set's the gridsquare from the record like this:
> 'COL_MY_GRIDSQUARE' => (!empty($record['my_gridsquare'])) ? $record['my_gridsquare'] : '',
>  'COL_MY_VUCC_GRIDS' => (!empty($record['my_vucc_grids'])) ? $record['my_vucc_grids'] : '',

But then it is overwritten from the station profile:
> if (strpos(trim($row['station_gridsquare']), ',') !== false) {
>                      $data['COL_MY_VUCC_GRIDS'] = strtoupper(trim($row['station_gridsquare']));
>                    } else {
>                      $data['COL_MY_GRIDSQUARE'] = strtoupper(trim($row['station_gridsquare']));
>                    }

I think this should be changed to only overwrite from the station profile if it does not exist in the imported data.
@magicbug agree?

**Reply by AndreasK79** [reply]
Thinking a bit more about this, I think this is what you have station locations for, and that you are supposed to create a separate location for this operation.

**Reply by AD7Djoe** [reply]
Yes, I think you are correct here ... I have done this before when I was in a different DXCC entity, but not for my Finnish proper call.  But, I deleted all the entries, created a new station location, then imported again.   Now, since my station location had my wanted grid defined, I ended up with proper logs.

This is a bit cumbersome though - one could argue that if the station call was the same (not a different dxcc entity), my suggestion could apply in the code.

**Reply by AndreasK79** [reply]
I'm not against changing this, and it's not difficult to do, it's just that I think it defeats the purpose of having station locations.
With version 2, you can also see grouped statistics for the locations defined in a logbook.

If @magicbug agrees, I can change the code to use the one from the ADIF, and only overwrite if not found.

**Reply by T012m3n7oR** [reply]
Wouldn't it make more sense to error the ADIF import and state to the user the gridsquare in the file does not match the station gridsquare, verify correct station?

**Comment by magicbug** [comment]
MY_GRIDSQUARE from within Cloudlog comes from Station Locations this will not be changed

We could add some code to see when importing if the MY_GRID exists and check it against Locations but apart from that its totally ignored within the HRD_ table I need to delete it in fact.

Adding that code would mean that if the gridsquare wasn't in locations it would create a location to assign against the QSO, but changing it any other way would create all sorts of problems in other parts of Cloudlog.
